src/util/piped_process.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CPROVER_UTIL_PIPED_PROCESS_H
+#define CPROVER_UTIL_PIPED_PROCESS_H
+
+// TODO: Revisit buffer size and storage location.
+// Note that this may impact number of characters correctly communicated
+// by the pipe, to check/fix.
+#define BUFSIZE 2048
+
+class piped_processt
+{
+  enum class process_statet
+  {
+    NOT_CREATED,
+    CREATED,
+    STOPPED
+  };
+
+  char **split_command_args(const std::string &command);
+
+protected:
+  FILE *response_stream;
+  FILE *command_stream;
+  int pipe_input[2];
+  int pipe_output[2];
```

Comment:
:bulb: These streams and pipes are operating system specific implementation details. It might be cleaner to use the "PImpl" coding pattern to keep all the `#ifdef _WIN32` stuff in the `.cpp` file. This should: 

- Make the header file easier to read for programmers of other parts of the code base who just want to use this API.
- Mean that we can keep the windows specific includes in the `.cpp` file, if we need fields with types based on `windows.h` when we come to do the windows support. This should also reduce compile times for the rest of the code base by avoiding putting `windows.h` in our header file.

This seems like a reasonable introduction to the code pattern if you aren't familiar with it - https://en.cppreference.com/w/cpp/language/pimpl We should be able to just leave off the `std::experimental::propagate_const` stuff it mentions.